src/main/scala/com.snowplowanalytics.iglu/client/Resolver.scala

```diff
@@ -90,75 +90,68 @@ object Resolver {
    *        resolver
    * @return a configured Resolver instance
    */
-  def apply(cacheSize: Int, refs: RepositoryRef*): Resolver =
-    Resolver(cacheSize, List[RepositoryRef](refs: _*))
-
-  /**
-   * Creates a Resolver instance from a JValue.
-   *
-   * @param config The JSON containing the configuration
-   *        for this resolver
-   * @return a configured Resolver instance
-   */
-  def parse(config: JValue): ValidatedNelType[Resolver] =
-    parse(asJsonNode(config))
+  def apply[F[_]: Sync](cacheSize: Int, refs: RepositoryRef*): F[Resolver[F]] =
+    SchemaCache[F](cacheSize).map(cache => new Resolver(cache, List(refs: _*)))
+
+  final case class ResolverConfig(cacheSize: Int, cacheTtl: Option[Int], repositoryRefs: List[Json])
+
+  val resolverConfigDecoder: AccumulatingDecoder[ResolverConfig] =
+    new Decoder[ResolverConfig] {
+      override final def apply(c: HCursor): Result[ResolverConfig] = {
+        for {
+          cacheSize <- c.downField("cacheSize").as[Int]
+          cacheTtl  <- c.downField("cacheTtl").as[Option[Int]]
+          repos     <- c.downField("repositories").as[List[Json]]
+        } yield ResolverConfig(cacheSize, cacheTtl, repos)
+      }
+    }.accumulating
 
   /**
-   * Constructs a Resolver instance from a JsonNode.
+   * Constructs a Resolver instance from a Json.
    *
    * @param config The JSON containing the configuration
    *        for this resolver
    * @return a configured Resolver instance
    */
-  def parse(config: JsonNode): ValidatedNelType[Resolver] = {
+  def parse[F[_]: Sync](config: Json): F[Either[NonEmptyList[ProcessingMessage], Resolver[F]]] = {
+    import ValidatableCirceMethods._
+    import ValidationExceptions._
 
     // We can use the bootstrap Resolver for working
     // with JSON Schemas here.
-    implicit val resolver     = Bootstrap.Resolver
-    implicit lazy val formats = org.json4s.DefaultFormats
-
-    import ValidatableJsonMethods._
-
-    // Check it passes validation
-    config
-      .verifySchemaAndValidate(ConfigurationSchema, dataOnly = true)
-      .andThen { node =>
-        val json = fromJsonNode(node)
-        val cacheSize =
-          validatedField[Int]("cacheSize")(json).leftMap(_.map(_.toString.toProcessingMessage))
-        val cacheTtl =
-          validatedOptionalField[Int]("cacheTtl")(json)
-            .leftMap(_.map(_.toString.toProcessingMessage))
-        val repositoryRefs: ValidatedNelType[RepositoryRefs] =
-          (validatedField[List[JValue]]("repositories")(json)).fold(
-            f => f.map(_.toString.toProcessingMessage).invalid,
-            s => getRepositoryRefs(s)
-          )
-        (cacheSize, repositoryRefs, cacheTtl).mapN {
-          Resolver(_, _, _)
-        }
-      }
-      .leftMap(errors =>
-        errors.prepend("Resolver configuration failed JSON Schema validation".toProcessingMessage))
+    val resolver = Bootstrap.resolver[F]
+
+    resolver.flatMap { implicit resolver =>
```

Comment:
I would like to get rid of "implicit resolver" pattern. I think we abused Scala implicits a lot with it. I believe it is generally a bad practice to pass resources (such as `ExecutionContext`) implicitly.

Reply:
yup it is especially cumbersome as the creation of a resolver is now an effect